syntax = "proto3";
import public "profile.proto";
import "google/protobuf/timestamp.proto";


option objc_class_prefix = "ETG";
option java_multiple_files = true;
option java_package = "etg.auth";
option java_outer_classname = "EtgProto";
option csharp_namespace = "Etg.Auth";

package etg.auth;

enum LoginType {
    EMAIL = 0;
    COMPANY_ID = 1;
    ACCESS_TOKEN = 2;
}

message LoginRequest {
    LoginType login_type = 1;
    string email = 2;
    string hashed_passwd = 3;
    string company_id = 4;
    string token = 5;  // token sent to contact's mobile
    string access_token = 6;
    string captcha_id = 7;
    string captcha = 8;
}

message LoginReply {
    bool is_successful = 1;
    string message = 2;
    string access_token = 3;
    google.protobuf.Timestamp timestamp_expire_at = 4;
    etg.profile.UserProfile user = 5;
    etg.profile.CompanyProfile company = 6;
}

